ng2-components/ng2-alfresco-core/src/dialogs/folder.dialog.spec.ts

```diff
@@ -0,0 +1,273 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { async, TestBed } from '@angular/core/testing';
+import { ComponentFixture } from '@angular/core/testing';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatDialogModule, MatDialogRef } from '@angular/material';
+import { BrowserDynamicTestingModule } from '@angular/platform-browser-dynamic/testing';
+import { Observable } from 'rxjs/Rx';
+
+import { AlfrescoTranslationService, CoreModule } from '../../index';
+import { TranslationMock } from '../assets/translation.service.mock';
+import { NodesApiService } from '../services/nodes-api.service';
+import { NotificationService } from '../services/notification.service';
+import { TranslationService } from '../services/translation.service';
+import { FolderDialogComponent } from './folder.dialog';
+
+describe('FolderDialogComponent', () => {
+
+    let fixture: ComponentFixture<FolderDialogComponent>;
+    let component: FolderDialogComponent;
+    let translationService: TranslationService;
+    let nodesApi: NodesApiService;
+    let notificationService: NotificationService;
+    let dialogRef;
+
+    beforeEach(async(() => {
+        dialogRef = {
+            close: jasmine.createSpy('close')
+        };
+
+        TestBed.configureTestingModule({
+            imports: [
+                CoreModule,
+                MatDialogModule,
+                FormsModule,
+                ReactiveFormsModule,
+                BrowserDynamicTestingModule
+            ],
+            providers: [
+                { provide: MatDialogRef, useValue: dialogRef },
+                { provide: AlfrescoTranslationService, useClass: TranslationMock }
+            ]
+        });
+
+        // entryComponents are not supported yet on TestBed, that is why this ugly workaround:
+        // https://github.com/angular/angular/issues/10760
+        TestBed.overrideModule(BrowserDynamicTestingModule, {
+            set: {entryComponents: [ FolderDialogComponent ]}
+        });
+
+        TestBed.compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(FolderDialogComponent);
+        component = fixture.componentInstance;
+
+        nodesApi = TestBed.get(NodesApiService);
+        notificationService = TestBed.get(NotificationService);
+
+        translationService = TestBed.get(TranslationService);
+        spyOn(translationService, 'get').and.returnValue(Observable.of('message'));
+    });
+
+    describe('Edit', () => {
+
+        beforeEach(() => {
+            component.data = {
+                folder: {
+                    id: 'node-id',
+                    name: 'folder-name',
+                    properties: {
+                        ['cm:description']: 'folder-description'
+                    }
+                }
+            };
+            component.ngOnInit();
```

Comment:
ngOnInit could prevent the component clean up some things on destroy. Why not `detectChanges` to trigger the life-cycle?

Reply:
I was not aware there could be problems on destroy. Replaced ngOnInit with detectChanges.